File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for(int i=1; i<=number; i++){
            int a = count(i);
            if(a>limit){
                answer += power;
            }else{
                answer += a;
            }
        }
        return answer;
    }
    public int count(int N){
        int count = 0;
        for (int i = 1; i * i <= N; i++) {
            if (i * i == N) count++;
            else if (N % i == 0) count += 2;
        }
        return count;
    }
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    for i in range(1, number + 1):
        a = count(i)
        if a > limit:
            answer += power
        else:
            answer += a
    
    return answer

def count(N):
    count = 0
    i = 1
    while i * i <= N:
        if i * i == N:
            count += 1
        elif N % i == 0:
            count += 2
        i += 1
    return count
